### ISSUE TITLE
Why PageRetriever returns null in unexpected ways

### ISSUE THREAD

**Issue by JanLenoch** [Kentico/xperience-training-13#77]
### Brief bug description

For a dark magic reason, `PageRetriever` fails to return a root page from the content tree in certain situations without a reason.

### Repro steps

1. Place breakpoints into `NavigationRepository.GetContentTreeNavigation`, `NavigationRepository.GetSecondaryNavigation`, and `NavigationRepository.GetConventionalRoutingNavigation`.
2. Add conditions so that they get caught only when `RootDto` is `null`.
3. Into each of the said methods, add another breakpoint.
4. This time, add inverted conditions (when `RootDto != null`).
5. Observe that when invoked from within the navigation view component, the repo (with its `PageRetriever`) fails to serve the `RootDto`.

### Expected behavior

`RootDto` is served everytime, including the call from the navigation view component.

### Test environment

 - Platform/OS: [e.g. .NET Core 3.1, iOS, Windows]
 - Browser [e.g. Chrome, Safari, Edge Chromium, Edge]
 - Version [e.g. 81]

### Additional context

Add any other context about the problem here. If you found an issue where a similar bug has been reported, add a link here.

### Screenshots

Add links to screenshots, if you deem it appropriate.

### Issue expands upon

This issue expands upon another issue(s): #`issue number`

### Blockers

This issue is blocked by the following issue(s): #`issue number`
